toffy/normalize.py

```diff
@@ -701,9 +719,18 @@ def normalize_image_data(img_dir, norm_dir, pulse_height_dir, panel_info,
     for fov in img_fovs:
         # compute per-mass normalization constant
         pulse_height_fov = pulse_height_df.loc[pulse_height_df['fov'] == fov, :]
-        norm_vals = norm_func(pulse_height_fov['pulse_height_fit'].values)
         channels = pulse_height_fov['Target'].values
 
+        # suppress warnings that can be caused by passing 0 or inf values into norm_func
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            norm_vals = norm_func(pulse_height_fov['pulse_height_fit'].values)
+
+        # nan or inf norm_vals can appear by passing zeros to norm_func, force back to 0
+        norm_vals[norm_vals == np.inf] = 0
+        norm_vals[norm_vals == -np.inf] = 0
+        norm_vals[np.isnan(norm_vals)] = 0
+
```

Comment:
Kinda roundabout to suppress warnings and correct the invalid results after the calculation. Maybe consider changing `def log()` and `def exp()` in [create_objective_function()](https://github.com/angelolab/toffy/blob/e6549638c933c19d1c19adfeb3c4783f18e7a348/toffy/normalize.py#L104) to always return a 0 when x<=0. Then it won't try to use np.log incorrectly and no warnings raised.

Reply:
Sounds good! Was a bit hesitant playing around with the actual objective functions at first, but this should work!